conan/tools/premake/toolchain.py

```diff
@@ -0,0 +1,296 @@
+    for prj in premake.workspace.eachproject({{ workspace_context_name }}) do
+        project(prj.name)
+    {{ project_body | indent(indent_level, first=True) }}
+    end -- End of for prj loop
+    {% else %}
+    project "{{ name }}"
+    {{ project_body | indent(indent_level, first=True) }}
+    {% endif %}
+    """))
+
+    # Template for the content INSIDE a project block
+    _project_body_template = jinja_env.from_string(textwrap.dedent(
+        """\
+        {% if kind %}
+        kind "{{ kind }}"
+        {% endif %}
+        {% if flags %}
+        -- Project flags {{ "(global)" if is_global else "(specific)"}}
+    {{ flags | indent(indent_level, first=True) }}
+        {% endif %}
+        -- Add other project settings here
+    """))
+
+    def __init__(self, name, workspace_context_name) -> None:
+        self.name = name
+        self.workspace_context_name = workspace_context_name
+        self.is_global = self.name == "*"
+        self.kind = None
+        self.extra_cxxflags = []
+        self.extra_cflags = []
+        self.extra_ldflags = []
+        self.extra_defines = {}
+        self.disable = False
+
+    def _generate_body(self, current_indent_level):
+        """Generates the inner content of the project block."""
+        flags_content = _generate_flags(self) # Generate flags specific to this project
+
+        return self._project_body_template.render(
+            kind="None" if self.disable else self.kind,
+            flags=flags_content,
+            is_global=self.is_global, # Pass is_global for the comment
+            indent_level=INDENT_LEVEL,
+            indent_spaces=INDENT_SPACES,
+            current_indent_level=current_indent_level
+        )
+
+    def _generate(self, current_indent_level=0):
+        """Generates the full project block (header + body)."""
+        project_body_content = self._generate_body(current_indent_level)
+
+        return self._premake_project_template.render(
+            is_global=self.is_global,
+            name=self.name,
+            workspace_context_name=self.workspace_context_name,
+            project_body=project_body_content,
+            indent_level=INDENT_LEVEL
+        )
+
+
+class PremakeToolchain:
+    """
+    PremakeToolchain generator
+    """
+
+    filename = "conantoolchain.premake5.lua"
+    # Keep template indented correctly for Lua output
+    _premake_file_template = textwrap.dedent(
+        """\
+    #!lua
+    -- Conan auto-generated toolchain file
+    {% if has_conan_deps %}
+    -- Include conandeps.premake5.lua with Conan dependency setup
+    include("conandeps.premake5.lua")
+    {% endif %}
+
+    -- Base build directory
+    local locationDir = path.normalize("{{ build_folder }}")
+
+    -- Generate workspace configurations
+    {% for workspace in workspaces.values() %}
+    {{ workspace._generate() }}
+    {% endfor %}
+    """)
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._workspaces = {}
```

Comment:
It is not very clear why the organization of the toolchain in workspaces.
It seems workspaces would be a feature of the project not a feature for the toolchain, but I have little idea about premake, so probably I need to undersand better about the workspaces in premake